src/components/Table/GridTable.stories.tsx

```diff
@@ -1162,6 +1162,98 @@ export function SelectableRows() {
   );
 }
 
+export function SelectableChildrenRows() {
+  type HeaderRow = { kind: "header"; id: string; data: undefined };
+  type ParentRow = { kind: "parent"; id: string; data: { name: string } };
+  type ChildRow = { kind: "child"; id: string; data: { name: string } };
+  type GrandChildRow = { kind: "grandChild"; id: string; data: { name: string } };
+  type Row = HeaderRow | ParentRow | ChildRow | GrandChildRow;
+
+  const selectCol = selectColumn<Row>();
+
+  const nameCol: GridColumn<Row> = {
+    header: "Name",
+    parent: ({ name }) => name,
+    child: ({ name }) => name,
+    grandChild: ({ name }) => name,
+    mw: "160px",
+  };
+
+  return (
+    <>
+      <GridTable
+        columns={[collapseColumn<Row>(), selectCol, nameCol]}
+        style={{ rowHeight: "fixed" }}
+        rows={
+          [
+            simpleHeader,
+            {
+              kind: "parent",
+              id: "1",
+              data: { name: "Howard Stark" },
+              initSelected: true,
+              inferSelectedState: false,
+              children: [
+                {
+                  kind: "child" as const,
+                  id: "2",
+                  data: {
+                    name: "Tony Stark",
+                  },
+                  initSelected: false,
+                  children: [
+                    {
+                      kind: "grandChild" as const,
+                      id: "5",
+                      data: {
+                        name: "Morgan Stark",
+                      },
+                      initSelected: true,
+                    },
+                  ],
+                },
+              ],
+            },
+            {
+              kind: "parent",
+              id: "3",
+              data: { name: "Odin" },
+              initSelected: false,
+              inferSelectedState: false,
+              children: [
+                {
+                  kind: "child" as const,
+                  id: "4",
+                  data: {
+                    name: "Thor",
+                  },
+                  initSelected: true,
+                },
+                {
+                  kind: "child" as const,
+                  id: "6",
+                  data: {
+                    name: "Hela",
+                  },
+                  initSelected: true,
+                },
+                {
+                  kind: "child" as const,
+                  id: "7",
+                  data: {
+                    name: "Loki",
+                  },
+                  initSelected: true,
```

Comment:
Can you remove the Hela and Loki rows? Afaict they are the same as the Thor row and not adding anything new to the test case?